src/node_options.h

```diff
@@ -0,0 +1,157 @@
+                                 "!MD5:"                            \
+                                 "!PSK:"                            \
+                                 "!SRP:"                            \
+                                 "!CAMELLIA"
+#endif
+
+#define V8_DEFAULT_THREAD_POOL_SIZE 4;
+
+enum NodeOptionsFlags {
+  NODE_OPTION_ABORT_ON_UNCAUGHT_EXCEPTION,
+  NODE_OPTION_AT_SECURE,
+  NODE_OPTION_INTERACTIVE,
+  NODE_OPTION_NO_DEPRECATION,
+  NODE_OPTION_EXPERIMENTAL_MODULES,
+  NODE_OPTION_EXPERIMENTAL_VM_MODULES,
+  NODE_OPTION_EXPOSE_INTERNALS,
+  NODE_OPTION_EVAL_STRING,
+  NODE_OPTION_NO_FORCE_ASYNC_HOOKS_CHECKS,
+  NODE_OPTION_NO_WARNINGS,
+#if HAVE_OPENSSL
+  NODE_OPTION_OPENSSL_CERT_STORE,
+# if NODE_FIPS_MODE
+  NODE_OPTION_OPENSSL_ENABLE_FIPS,
+  NODE_OPTION_OPENSSL_FORCE_FIPS,
+# endif
+#endif
+  NODE_OPTION_PENDING_DEPRECATION,
+  NODE_OPTION_PRESERVE_SYMLINKS,
+  NODE_OPTION_PRINT_EVAL,
+  NODE_OPTION_PROFILING,
+  NODE_OPTION_PROF_PROCESS,
+  NODE_OPTION_SYNTAX_CHECK,
+  NODE_OPTION_THROW_DEPRECATION,
+  NODE_OPTION_TRACE_DEPRECATION,
+  NODE_OPTION_TRACE_ENABLED,
+  NODE_OPTION_TRACE_SYNC_IO,
+  NODE_OPTION_TRACE_WARNINGS,
+  NODE_OPTION_TRACK_HEAP_OBJECTS,
+  NODE_OPTION_ZERO_FILL_BUFFERS
+};
+
+enum NodeOptionsStrings {
+  NODE_OPTION_STRING_EVAL_STRING,
+#if defined(NODE_HAVE_I18N_SUPPORT)
+  NODE_OPTION_STRING_ICU_DATA_DIR,
+#endif
+#if HAVE_OPENSSL
+  NODE_OPTION_STRING_OPENSSL_CONFIG,
+  NODE_OPTION_STRING_OPENSSL_DEFAULT_CIPHER_LIST,
+  NODE_OPTION_STRING_OPENSSL_EXTRA_CA_CERTS,
+#endif
+  NODE_OPTION_STRING_TRACE_CATEGORIES,
+  NODE_OPTION_STRING_USERLAND_LOADER,
+  NODE_OPTION_STRING_WARNING_FILE,
+  NODE_OPTION_STRING_COUNT
+};
+
+// The node::NodeOptions class encapsulates startup and options details
+// about the Node.js process. The NodeOptions object is created within
+// node_main.cc and passed down to collect current state detail.
+class NodeOptions {
+ public:
+  static void ParseArgs(int* argc,
+                        const char** argv,
+                        int* exec_argc,
+                        const char*** exec_argv,
+                        int* v8_argc,
+                        const char*** v8_argv,
+                        bool is_env,
+                        DebugOptions* debug_options,
+                        NodeOptions* state);
+
+  NodeOptions();
+  ~NodeOptions() {}
+
+  bool SafeGetenv(const char* key, std::string* text);
+
+  int v8_thread_pool_size() const { return v8_thread_pool_size_; }
+  uint64_t prog_start_time() const { return prog_start_time_; }
+  std::vector<std::string> preload_modules() const { return preload_modules_; }
+
+  void SetProgStartTime() {
+    prog_start_time_ = uv_now(uv_default_loop());
+  }
+
+  void SetRevert(int cve) {
+    reverted_ |= (1 << cve);
+  }
+
+  bool IsReverted(int cve) {
+    return reverted_ & (1 << cve);
+  }
+
+  bool IsSet(NodeOptionsFlags flag);
+  std::string GetString(NodeOptionsStrings option);
```

Comment:
This could be `const std::string& GetString(NodeOptionsStrings option) const;`, so we don't need to make copies every time this is called

Reply:
Ack on all points :)